### ISSUE TITLE
Thunderbird fails to find the settings for my email accound

### ISSUE THREAD

**Issue by quisys** [LukeSmithxyz/emailwiz#73]
I followed the video step-by-step, but thunderbird doesn't connect to my email server

journalctl -xe seems to be in order
```
-- 
-- A start job for unit UNIT has finished successfully.
-- 
-- The job identifier is 3.
Sep 02 16:06:52 ramble systemd[696]: Reached target Basic System.
-- Subject: A start job for unit UNIT has finished successfully
-- Defined-By: systemd
-- Support: https://www.debian.org/support
-- 
-- A start job for unit UNIT has finished successfully.
-- 
-- The job identifier is 2.
Sep 02 16:06:52 ramble systemd[1]: Started User Manager for UID 1000.
-- Subject: A start job for unit user@1000.service has finished successfully
-- Defined-By: systemd
-- Support: https://www.debian.org/support
-- 
-- A start job for unit user@1000.service has finished successfully.
-- 
-- The job identifier is 447.
Sep 02 16:06:52 ramble systemd[1]: Started Session 1 of user finn.
-- Subject: A start job for unit session-1.scope has finished successfully
-- Defined-By: systemd
-- Support: https://www.debian.org/support
-- 
-- A start job for unit session-1.scope has finished successfully.
-- 
-- The job identifier is 503.
Sep 02 16:06:52 ramble systemd[696]: Reached target Default.
-- Subject: A start job for unit UNIT has finished successfully
-- Defined-By: systemd
-- Support: https://www.debian.org/support
-- 
-- A start job for unit UNIT has finished successfully.
-- 
-- The job identifier is 1.
Sep 02 16:06:52 ramble systemd[696]: Startup finished in 80ms.
-- Subject: User manager start-up is now complete
-- Defined-By: systemd
-- Support: https://www.debian.org/support
-- 
-- The user manager instance for user 1000 has been started. All services queued
-- for starting have been started. Note that other services might still be starting
-- up or be started at any later time.
-- 
-- Startup of the manager took 80429 microseconds.
Sep 02 16:11:43 ramble sudo[733]:     finn : TTY=pts/0 ; PWD=/home/finn ; USER=root ; COMMAND=/usr/sbin/usermod -a -G systemd-journal finn
Sep 02 16:11:43 ramble sudo[733]: pam_unix(sudo:session): session opened for user root by finn(uid=0)
Sep 02 16:11:43 ramble usermod[734]: add 'finn' to group 'systemd-journal'
Sep 02 16:11:43 ramble usermod[734]: add 'finn' to shadow group 'systemd-journal'
Sep 02 16:11:43 ramble sudo[733]: pam_unix(sudo:session): session closed for user root
Sep 02 16:11:52 ramble su[740]: (to finn) finn on pts/0
Sep 02 16:11:52 ramble su[740]: pam_unix(su:session): session opened for user finn by finn(uid=1000)
Sep 02 16:16:51 ramble sshd[751]: Did not receive identification string from 223.17.149.122 port 60852
Sep 02 16:16:52 ramble sshd[752]: Did not receive identification string from 62.212.43.129 port 60727
Sep 02 16:17:01 ramble CRON[754]: pam_unix(cron:session): session opened for user root by (uid=0)
Sep 02 16:17:01 ramble CRON[755]: (root) CMD (   cd / && run-parts --report /etc/cron.hourly)
Sep 02 16:17:01 ramble CRON[754]: pam_unix(cron:session): session closed for user root
Sep 02 16:17:02 ramble sudo[757]:     finn : TTY=pts/0 ; PWD=/home/finn ; USER=root ; COMMAND=/bin/journalctl -xe
Sep 02 16:17:02 ramble sudo[757]: pam_unix(sudo:session): session opened for user root by finn(uid=0)
```

what's going on?

**Comment by juandpineiro** [comment]
@quisys check the contents of your `/var/log/mail.log`. I was having maybe the same kind of issue and the log just reported the connection attempts (from thunderbird) and immediate disconnects.

Check the output of the `doveadm` command. If the result is an authentication error or an error in the `dovecot.conf` file, then you probably have the same issue as I had.

The `/etc/dovecot.conf` file from the `emailwiz` script contains a line with `ssl_dh = ...` which is not supported by dovecot < 2.3. So you will have to check your dovecot version and upgrade it to 2.3.

Also you'll need to generate de Diffie-Hellman params in `/usr/share/dovecot/dh.pem`.

After following those steps and restarting `postfix` and `dovecot` the output of your `doveadm` command should be a common usage prompt. Then try to connect from Thunderbird again and it should work.

TLDR: The script assumes `dovecot > 2.3` is installed and the final configuration is left without ssl authentication set-up if you happen to have dovecot < 2.3 installed.

**Comment by acagastya** [comment]
I am having the same issue.  Following what @juandpineiro said, I checked `/etc/dovecot.conf` and that file does not exist.

**Comment by juandpineiro** [comment]
@acagastya if that's the case you have to generate the `dh.pem` file manually with the `openssl` command: 

```
# Note that this command will take a LONG time and use 100% of a CPU core
openssl dhparam 4096 > /usr/share/dovecot/dh.pem
```

**Comment by acagastya** [comment]
@juandpineiro what kind of firewall configuration is required for running this?  I started over, but this time, I am facing `Timeout during connect (likely firewall problem)` on Debian.

**Comment by acagastya** [comment]
I have `dovecot v2.3.4.1`, yet the Thunderbird does not connect to the server.

**Comment by acagastya** [comment]
@quisys I think it is due to firewall.  Try running `journalctl -xe`.  Do you see `[UFW BLOCK]`?  It is probably because of firewall.

**Comment by user234683** [comment]
@acagastya Thanks, that was my problem. Instead of picking the ports to allow manually, I used `ufw app list` to view available profiles. Then I used `ufw allow 'Dovecot IMAP'`, `ufw allow 'Dovecot Secure IMAP'`, etc. for all the profiles listed for Dovecot and Postfix

**Comment by kobold2301** [comment]
Hello,
it seems like i have the same problem, but allowing ufw allow 'Dovecot IMAP', ufw allow 'Dovecot Secure IMAP' did not change anything.

I am using Linode and the support made lifted the SMTP port restrictions. So everything should be fine, but still i can not log in into Thunderbird.

Thanks for the help already :)



running : sudo netstat -tulpn | grep LISTEN, gives this output
tcp        0      0 0.0.0.0:25              0.0.0.0:*               LISTEN      15087/master        
tcp        0      0 0.0.0.0:443             0.0.0.0:*               LISTEN      14721/nginx: master 
tcp        0      0 127.0.0.1:8000          0.0.0.0:*               LISTEN      12019/python3       
tcp        0      0 0.0.0.0:587             0.0.0.0:*               LISTEN      15087/master        
tcp        0      0 127.0.0.1:12301         0.0.0.0:*               LISTEN      14883/opendkim      
tcp        0      0 127.0.0.1:783           0.0.0.0:*               LISTEN      14861/perl          
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      14721/nginx: master 
tcp        0      0 0.0.0.0:465             0.0.0.0:*               LISTEN      15087/master        
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN      14926/systemd-resol 
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      2316/sshd           
tcp6       0      0 :::25                   :::*                    LISTEN      15087/master        
tcp6       0      0 :::587                  :::*                    LISTEN      15087/master        
tcp6       0      0 ::1:783                 :::*                    LISTEN      14861/perl          
tcp6       0      0 :::80                   :::*                    LISTEN      14721/nginx: master 
tcp6       0      0 :::465                  :::*                    LISTEN      15087/master        
tcp6       0      0 :::22                   :::*                    LISTEN      2316/sshd   


running: sudo journalctl -xe:
Apr 15 09:22:49 flask-blumenhaus sshd[15399]: Received disconnect from 49.88.112.72 port 41168:11:  [preauth]
Apr 15 09:22:49 flask-blumenhaus sshd[15399]: Disconnected from 49.88.112.72 port 41168 [preauth]
Apr 15 09:23:13 flask-blumenhaus sshd[15401]: Accepted publickey for ludw from 77.23.254.111 port 58460 ssh2: RSA SHA256:bpaSxJRX8iNalzVDLtc6+Omwg0Z+iNzhe6XXxo/IuCM
Apr 15 09:23:16 flask-blumenhaus kernel: [UFW BLOCK] IN=eth0 OUT= MAC=f2:3c:92:2b:c1:dd:50:87:89:40:a1:c1:08:00 SRC=159.203.242.122 DST=172.105.249.246 LEN=40 TOS=0x00 PREC=0
Apr 15 09:23:27 flask-blumenhaus kernel: [UFW BLOCK] IN=eth0 OUT= MAC=f2:3c:92:2b:c1:dd:50:87:89:40:a1:c1:08:00 SRC=205.185.114.54 DST=172.105.249.246 LEN=44 TOS=0x00 PREC=0x
Apr 15 09:23:27 flask-blumenhaus sudo[15419]:     ludw : TTY=pts/0 ; PWD=/home/ludw ; USER=root ; COMMAND=/bin/netstat -tulpn
Apr 15 09:23:27 flask-blumenhaus sudo[15419]: pam_unix(sudo:session): session opened for user root by ludw(uid=0)
Apr 15 09:23:27 flask-blumenhaus sudo[15419]: pam_unix(sudo:session): session closed for user root
Apr 15 09:23:40 flask-blumenhaus kernel: [UFW BLOCK] IN=eth0 OUT= MAC=f2:3c:92:2b:c1:dd:e4:c7:22:1f:22:c1:08:00 SRC=171.67.70.87 DST=172.105.249.246 LEN=40 TOS=0x00 PREC=0x00
Apr 15 09:23:42 flask-blumenhaus sshd[15422]: Invalid user os from 42.200.109.74 port 56520
Apr 15 09:23:42 flask-blumenhaus sshd[15422]: Received disconnect from 42.200.109.74 port 56520:11: Bye Bye [preauth]
Apr 15 09:23:42 flask-blumenhaus sshd[15422]: Disconnected from invalid user os 42.200.109.74 port 56520 [preauth]
Apr 15 09:23:54 flask-blumenhaus kernel: [UFW BLOCK] IN=eth0 OUT= MAC=f2:3c:92:2b:c1:dd:50:87:89:40:a1:c1:08:00 SRC=165.227.94.27 DST=172.105.249.246 LEN=40 TOS=0x00 PREC=0x0
Apr 15 09:24:14 flask-blumenhaus kernel: [UFW BLOCK] IN=eth0 OUT= MAC=f2:3c:92:2b:c1:dd:50:87:89:40:a1:c1:08:00 SRC=74.120.14.88 DST=172.105.249.246 LEN=44 TOS=0x00 PREC=0x20
Apr 15 09:24:23 flask-blumenhaus sudo[15424]:     ludw : TTY=pts/0 ; PWD=/home/ludw ; USER=root ; COMMAND=/bin/journalctl -xe
Apr 15 09:24:23 flask-blumenhaus sudo[15424]: pam_unix(sudo:session): session opened for user root by ludw(uid=0)

running this: sudo ufw status:
Status: active
To                         Action      From
--                         ------      ----
22/tcp                     ALLOW       Anywhere                  
80/tcp                     ALLOW       Anywhere                  
443/tcp                    ALLOW       Anywhere                  
25/tcp                     ALLOW       Anywhere                  
Dovecot IMAP               ALLOW       Anywhere                  
Dovecot Secure IMAP        ALLOW       Anywhere                  
25                         ALLOW       Anywhere                  
443                        ALLOW       Anywhere                  
587                        ALLOW       Anywhere                  
783                        ALLOW       Anywhere                  
80                         ALLOW       Anywhere                  
465                        ALLOW       Anywhere                  
22                         ALLOW       Anywhere                  
993                        ALLOW       Anywhere                  
Anywhere                   ALLOW       127.0.0.1                 
578/tcp                    ALLOW       Anywhere                  
587/tcp                    ALLOW       Anywhere                  
993/tcp                    ALLOW       Anywhere                  
Postfix                    ALLOW       Anywhere                  
Nginx HTTP                 ALLOW       Anywhere                  
Nginx HTTPS                ALLOW       Anywhere                  
22/tcp (v6)                ALLOW       Anywhere (v6)             
80/tcp (v6)                ALLOW       Anywhere (v6)             
443/tcp (v6)               ALLOW       Anywhere (v6)             
25/tcp (v6)                ALLOW       Anywhere (v6)             
Dovecot IMAP (v6)          ALLOW       Anywhere (v6)             
Dovecot Secure IMAP (v6)   ALLOW       Anywhere (v6)             
25 (v6)                    ALLOW       Anywhere (v6)             
443 (v6)                   ALLOW       Anywhere (v6)             
587 (v6)                   ALLOW       Anywhere (v6)             
783 (v6)                   ALLOW       Anywhere (v6)             
80 (v6)                    ALLOW       Anywhere (v6)             
465 (v6)                   ALLOW       Anywhere (v6)             
22 (v6)                    ALLOW       Anywhere (v6)             
993 (v6)                   ALLOW       Anywhere (v6)             
578/tcp (v6)               ALLOW       Anywhere (v6)             
587/tcp (v6)               ALLOW       Anywhere (v6)             
993/tcp (v6)               ALLOW       Anywhere (v6)             
Postfix (v6)               ALLOW       Anywhere (v6)             
Nginx HTTP (v6)            ALLOW       Anywhere (v6)             
Nginx HTTPS (v6)           ALLOW       Anywhere (v6)
